react/components/Table/Toolbar/index.js

```diff
@@ -0,0 +1,167 @@
+import React, { Fragment } from 'react'
+import PropTypes from 'prop-types'
+
+import InputToolbar from './InputToolbar'
+import ButtonDensity from './ButtonDensity'
+import ButtonFields from './ButtonFields'
+import ButtonDownload from './ButtonDownload'
+import ButtonUpload from './ButtonUpload'
+import ButtonNewLine from './ButtonNewLine'
+import ExtraActions from './ExtraActions'
+import ButtonToolbar from './ButtonToolbar'
+
+const Container = ({ justify = 'end', children }) => (
+  <div id="toolbar" className={`mb5 flex flex-row w-100 justify-${justify}`}>
+    {children}
+  </div>
+)
+
+Container.propTypes = {
+  justify: PropTypes.oneOf(['between', 'end', 'start', 'around', 'center']),
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+}
+
+const Toolbar = ({
+  actions: {
+    inputSearch,
+    download,
+    upload,
+    fields,
+    extraActions,
+    newLine,
+    density,
+  },
+  loading,
+  children,
+}) => {
+  const isDownloadVisible = download && download.handleCallback
+  const isUploadVisible = upload && upload.handleCallback
+  const isFieldsVisible = fields && fields.showAllLabel && fields.hideAllLabel
+  const isExtraActionsVisible =
+    extraActions && extraActions.label && extraActions.actions.length > 0
+  const isNewLineVisible = newLine && newLine.label
+  const isSearchBarVisible = !!inputSearch
+  const isDensityVisible =
+    density &&
+    density.buttonLabel &&
+    density.lowOptionLabel &&
+    density.mediumOptionLabel &&
+    density.highOptionLabel
+
+  return (
+    children || (
+      <Container justify={isSearchBarVisible ? 'between' : 'end'}>
+        {inputSearch && (
+          <InputToolbar disabled={loading} inputSearch={inputSearch} />
+        )}
+        <div className="flex flex-row items-center">
+          <Fragment>
+            {isDensityVisible && (
+              <ButtonDensity density={density} disabled={loading} />
+            )}
+            {isFieldsVisible && (
+              <ButtonFields fields={fields} disabled={loading} />
+            )}
+            {isDownloadVisible && (
+              <ButtonDownload download={download} disabled={loading} />
+            )}
+            {isUploadVisible && (
+              <ButtonUpload upload={upload} disabled={loading} />
+            )}
+            {isExtraActionsVisible && (
+              <ExtraActions extraActions={extraActions} />
+            )}
+            {isNewLineVisible && (
+              <ButtonNewLine newLine={newLine} disabled={loading} />
+            )}
+          </Fragment>
+        </div>
+      </Container>
+    )
+  )
+}
+
+Toolbar.Container = Container
```

Comment:
@estacioneto  here is the answer that you looking for. The `Toolbar` is a compound component. This allows the usage  
```js
<Table>
  <Table.Toolbar>
    <Table.Toolbar.Container>
      <Table.Toolbar.ButtonDensity {...} />
      <Table.Toolbar.ButtonDownload {...} />
      <Table.Toolbar.ETC {...} />
    </Table.Toolbar.Container>
  </Table.Toolbar>
</Table>
```

Reply:
@Klynger, take a look as well :)